be/src/vec/functions/function_running_difference.h

```diff
@@ -0,0 +1,188 @@
+                prev = src[i];
+                has_prev_value = true;
+            }
+            else
+            {
+                auto cur = src[i];
+                /// Overflow is Ok.
+                dst[i] = static_cast<Dst>(cur) - prev;
+                prev = cur;
+            }
+        }
+    }
+
+    /// Result type is same as result of subtraction of argument types.
+    template <typename SrcFieldType>
+    using DstFieldType = typename NumberTraits::ResultOfSubtraction<SrcFieldType, SrcFieldType>::Type;
+
+    /// Call polymorphic lambda with tag argument of concrete field type of src_type.
+    template <typename F>
+    void dispatchForSourceType(const IDataType & src_type, F && f) const
+    {
+        WhichDataType which(src_type);
+
+        if (which.is_uint8())
+            f(UInt8());
+        else if (which.is_uint16())
+            f(UInt16());
+        else if (which.is_uint32())
+            f(UInt32());
+        else if (which.is_uint64())
+            f(UInt64());
+        else if (which.is_int8())
+            f(Int8());
+        else if (which.is_int16())
+            f(Int16());
+        else if (which.is_int32())
+            f(Int32());
+        else if (which.is_int64())
+            f(Int64());
+        else if (which.is_float32())
+            f(Float32());
+        else if (which.is_float64())
+            f(Float64());
+        else if (which.is_date())
+            f(DataTypeDate::FieldType());
+        else if (which.is_date_v2())
+            f(DataTypeDateV2::FieldType());
+        else if (which.is_date_time())
+            f(DataTypeDateTime::FieldType());
+        else
+            throw Exception("Argument for function " + get_name() + " must have numeric type.", 1/*ErrorCode::ILLEGAL_TYPE_OF_ARGUMENT*/);
+    }
+
+public:
+    static constexpr auto name = "running_difference";
+
+    static FunctionPtr create(){
+        return std::make_shared<FunctionRunningDifference>();
+    }
+
+    String get_name() const override {
+        return name;
+    }
+
+    bool is_stateful() const override{
+        return true;
+    } 
+
+    size_t get_number_of_arguments() const override
+    {
+        return 1;
+    }
+
+    bool is_deterministic() const override { return false; }
+    bool is_deterministic_in_scope_of_query() const override
+    {
+         return false;
+    } 
+
+   // bool is_suitable_for_short_circuit_arguments_execution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes & arguments) const override
+    {
+        DataTypePtr res;
+        dispatchForSourceType(*remove_nullable(arguments[0]), [&](auto field_type_tag)
+        {
+            res = std::make_shared<DataTypeNumber<DstFieldType<decltype(field_type_tag)>>>();
+        });
+
+        if (arguments[0]->is_nullable())
+            res = make_nullable(res);
+
```

Comment:
I see in ...functions.py file, you set return type is ALWAYS_NOT_NULLABLE, but here make_nullable